flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/session/SessionManagerImpl.java

```diff
@@ -68,13 +73,16 @@ public class SessionManagerImpl implements SessionManager {
     private @Nullable ScheduledExecutorService cleanupService;
     private @Nullable ScheduledFuture<?> timeoutCheckerFuture;
 
+    private CatalogStoreFactory catalogStoreFactory;
+
     public SessionManagerImpl(DefaultContext defaultContext) {
         this.defaultContext = defaultContext;
         ReadableConfig conf = defaultContext.getFlinkConfig();
         this.idleTimeout = conf.get(SQL_GATEWAY_SESSION_IDLE_TIMEOUT).toMillis();
         this.checkInterval = conf.get(SQL_GATEWAY_SESSION_CHECK_INTERVAL).toMillis();
         this.maxSessionCount = conf.get(SQL_GATEWAY_SESSION_MAX_NUM);
         this.sessions = new ConcurrentHashMap<>();
+        this.catalogStoreFactory = initCatalogStoreFactory(defaultContext);
```

Comment:
From my understanding, all resource initializing should happen in open/start method.